cmake/IRODSConfigVersion.cmake.in

```diff
@@ -0,0 +1,158 @@
+# CMake package version file for iRODS, for use with find_package().
+#
+# A couple of details about our version nubmers:
+#
+# Prior to iRODS version 5.0, major releases incremented the second part of the version number,
+# and minor releases incremented the third part. For example, 4.3.0 was a major release, and 4.3.1
+# was a minor release of 4.3.
+# Starting with iRODS 5.0, major releases increment the first part of the version number, and
+# minor releases increment the second part. For example, 5.0 is a major release, and 5.1 is a
+# minor release of 5. Patch releases, should we decide to do them, increment the third part of the
+# version number. For example, 5.0.1 would be a patch release of 5.0.
+# 4.3 was the last major release before 5.0.
+#
+# Additionally, starting with the leadup to iRODS version 5.0, we have adopted .9x version numbers
+# for development builds. For example, 4.97 would be a development version between 4.3 and 5.0,
+# and 5.92 would be a development version between 5 and 6.0.
+# Starting with iRODS 5.0, this extends to minor releases as well. For example, 5.1.94 would be a
+# development version between 5.1 and 5.2.0.
+#
+# Thusly, our CMake package version file is designed as so:
+# - An iRODS installation should match as exactly compatible if the find_package()-calling project
+#   is searching for the exact same version.
+# - An iRODS installation should match as compatible if the find_package()-calling project is
+#   searching for a version range containing the version of the iRODS installation.
+# - An iRODS installation without a .9x version should match as potentially compatible if the
+#   find_package()-calling project is searching for an earlier version of the same major release
+#   or a .9x version of the previous major release.
+# - An iRODS installation with a .9x version in the second part of the version number should match
+#   as potentially compatible if the find_package()-calling project is searching for the next
+#   major release or any earlier version of the same major release.
+# - An iRODS installation with a .9x version in the third part of the version number should match
+#   as potentially compatible if the find_package()-calling project is searching for the next
+#   minor release or any earlier version of the same major release.
```

Comment:
Does this only matter if the consuming project passes version arguments to the  `find_package` call?

For example, if version arguments aren't passed to `find_package`, can we assume the project is going to adapt its build/implementation based on the version information returned by this cmake module?

Reply:
> Does this only matter if the consuming project passes version arguments to the `find_package` call?

Correct. If no version number or range is passed to `find_package`, the only thing checked is bit-ness.

> For example, if version arguments aren't passed to `find_package`, can we assume the project is going to adapt its build/implementation based on the version information returned by this cmake module?

Yes.